plugins/Sid/resid/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+# These are the defaults
+set(RESID_INLINING 1)
+set(RESID_INLINE inline)
+set(RESID_BRANCH_HINTS 1)
+set(NEW_8580_FILTER 1)
+
+set(HAVE_BOOL 1)
+set(HAVE_LOG1P 1)
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "GCC,Clang")
+    set(HAVE_BUILTIN_EXPECT 1)
+else()
+    set(HAVE_BUILTIN_EXPECT 0)
+endif()
+
+configure_file(resid/siddefs.h.in resid/siddefs.h @ONLY)
+configure_file(resid/siddefs.h.in ${CMAKE_CURRENT_SOURCE_DIR}/resid/siddefs.h @ONLY)
```

Comment:
> Not sure why there need to be two configured copies of this file. Also, I would generally avoid writing to the source directory when configuring, especially to Git repositories where the target file isn't ignored.

I believe ``siddefs.h`` was put in the build directory with ``configure_file(resid/siddefs.h.in resid/siddefs.h @ONLY)``, which caused the builds to fail because it couldn't be found when it was included from within the source files. I think the expectation is for that file to end up in the source directory, where the other files are, as this was where ``siddefs.h`` was located even for the [current submodule](https://github.com/simonowen/resid/blob/master/siddefs.h).

For keeping ``configure_file(resid/siddefs.h.in resid/siddefs.h @ONLY)``, I wasn't sure if putting it in the build directory was still needed, but I think I can do away with it if I keep the one that puts the file in the source directory. Or, if there is another way I can make the source files find ``siddefs.h``, I'll go with that.

Reply:
Can't you just do `target_include_directories(resid PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/resid")`? (Or `resid_objs` if you follow one of the later suggestions.)

Reply:
> Can't you just do target_include_directories(resid PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/resid")? (Or resid_objs if you follow one of the later suggestions.)

For sure, just thought the ``configure_file`` should do the same job, although you're probably right that we shouldn't touch the source directory.